### PR TITLE
Testing refactoring

### PR DESCRIPTION
**Pull request by ElvinEfendi**
I've fixed problems with test environment setup and configured the gem to use minitest for testing. Then added specifications that covers core flow such as fetching the config and storing it. Moreover I did some code refactoring on the way.

@Shopify/traffic

### CODE REVIEW

**Comment by ElvinEfendi** [comment]
@ibawt and @charlescng can you review this?

**Comment by ibawt** [comment]
LGTM.  Nice job on the code quality.

File: lib/oxidized/string.rb [hunk 1]
```diff
@@ -16,7 +16,7 @@ def cut_head
     # sets @cmd and @name unless @name is already set
     def set_cmd command
       @cmd  = command
-      @name = @cmd.strip.gsub(/\s+/, '_') if @name == nil
+      @name ||= @cmd.strip.gsub(/\s+/, '_')
```
Comment (ibawt) [inline lib/oxidized/string.rb:19]: +1

File: oxidized.gemspec [hunk 2]
```diff
@@ -1,6 +1,11 @@
+# coding: utf-8
+lib = File.expand_path('../lib', __FILE__)
+$LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
+require 'oxidized/version'
+
 Gem::Specification.new do |s|
   s.name              = 'oxidized'
-  s.version           = '0.8.1'
+  s.version           = Oxidized::VERSION
```
Comment (ibawt) [inline oxidized.gemspec:8]: thanks for this as well

**Comment by charlescng** [comment]
LGTM. Can we setup Travis CI as well?

**Comment by ElvinEfendi** [comment]
@charlescng test specs assumes that there is ~/.config/oxidized/routers.db file with list of at least 3 online network devices that are accessible via ssh from the host machine. I do not know how to handle this in Travis. Any suggestion?

**Comment by ElvinEfendi** [comment]
I tried to make the code base more test friendly and integrated Travic CI.  @ibawt @charlescng @marc-barry

**Comment by ibawt** [comment]
cool I think it's time to merge

**Pull request merged**